federation/pkg/federatedtypes/crudtester/crudtester.go

```diff
@@ -139,11 +144,11 @@ func (c *FederatedTypeCRUDTester) CheckDelete(obj pkgruntime.Object, orphanDepen
 
 	deletingInCluster := (orphanDependents != nil && *orphanDependents == false)
 
-	waitTimeout := wait.ForeverTestTimeout
-	if deletingInCluster {
-		// May need extra time to delete both federation and cluster resources
-		waitTimeout = c.clusterWaitTimeout
-	}
+	waitTimeout := ForeverTimeout
```

Comment:
Pass the right clusterWaitTimeout in https://github.com/kubernetes/kubernetes/blob/50ec4384ff0ac482beb164767e562765ad82bde7/test/integration/federation/framework/crudtester.go#L46 rather than overwriting it here.